proposals/0000-add-support-for-display-p3-colors.md

```diff
@@ -0,0 +1,209 @@
+
+    // handle json with srgb color space specified
+    if ((value = [dictionary objectForKey:@"srgb"])) {
+      return [RCTConvert @[value.red, value.green, value.blue, value.alpha]];
+
+    // handle json with display-p3 color space specified
+    } else if ((value = [dictionary objectForKey:@"display-p3"])) {
+      return [UIColor colorWithDisplayP3Red:value.red green:value.green blue:value.blue alpha:value.alpha];
+
+    // ...
+```
+
+2. Update [ColorComponents.h](https://github.com/facebook/react-native/blob/528f97152b7e0a7465c5b5c02e96c2c4306c78fe/packages/react-native/ReactCommon/react/renderer/graphics/ColorComponents.h) to include color space.
+
+```cpp
+enum class ColorSpace {
+  sRGB,
+  DisplayP3
+};
+
+struct ColorComponents {
+  float red{0};
+  float green{0};
+  float blue{0};
+  float alpha{0};
+  ColorSpace colorSpace{ColorSpace::sRGB};  // Default to sRGB
+};
+```
+
+3. Update [RCTConversions](https://github.com/facebook/react-native/blob/16ad818d21773cdf25156642fae83592352ae534/packages/react-native/React/Fabric/RCTConversions.h#L37) to handle setting new color values for color space. If color space is not included preserve existing behavior.
+
+4. Update [RCTTextPrimitivesConversions](https://github.com/facebook/react-native/blob/ac1cdaa71620d5bb4860237cafb108f6aeae9aef/packages/react-native/ReactCommon/react/renderer/textlayoutmanager/platform/ios/react/renderer/textlayoutmanager/RCTTextPrimitivesConversions.h#L116) to handle setting new color values for color space. If color space is not included preserve existing behavior.
+
+```cpp
+auto components = facebook::react::colorComponentsFromColor(sharedColor);
+if (components.colorSpace == ColorSpace::DisplayP3) {
+  return [UIColor colorWithDisplayP3Red:components.red green:components.green blue:components.blue alpha:components.alpha];
+} else {
+  return [UIColor colorWithRed:components.red green:components.green blue:components.blue alpha:components.alpha];
+}
+```
+
+### Android Changes
+
+1. Enable the wide color gamut in [ReactActivity](https://github.com/facebook/react-native/blob/7625a502960e6b107e77542ff0d6f40fbf957322/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactActivity.java#L22) if available.
+
+```java
+Window window = getWindow();
+boolean isScreenWideColorGamut = window.getDecorView().isScreenWideColorGamut()
+if (isScreenWideColorGamut) {
+  window.setColorMode(PixelFormat.COLOR_MODE_WIDE_COLOR_GAMUT)
+}
+```
+
+This is prerequisite per [Android documentation](https://developer.android.com/training/wide-color-gamut) but they note:
+
+> When wide color gamut mode is enabled, the activity's window uses more memory and GPU processing for screen composition. Before enabling wide color gamut mode, you should carefully consider if the activity truly benefits from it. For example, an activity that displays photos in fullscreen is a good candidate for wide color gamut mode, but an activity that shows small thumbnails is not.
```

Comment:
Given this, we might have to run experiments internally at Meta.
This means that we might have to add a feature flag rather than have the Activity enable the WideGamut by default if the device supports that.
 I don't have the technical details of which code changes are required right now. But we can move on with the proposal as this is an implementation details which is Meta-specific, and I'll reach out to Android people to understand how to do it.